### PR TITLE
Expand docs

### LINKED ISSUE #143: Expand API Documentation
Create separate pages for each api endpoint, with proper examples on how to use the APIs (including URLs that the user can test).

Can also add documentation for the types returned; specifically, the schema (see [v3.3.2](https://github.com/qbreader/website/releases/tag/v3.3.2)) of the tossup and bonus documents returned can also be documented.

### PR DESCRIPTION
**Pull request by g3ner1c**
Closes #143 when completed

[click for pr summary](https://github.com/qbreader/website/pull/150#issuecomment-1520591437)

### CODE REVIEW

**Comment by g3ner1c** [comment]
RFC on this new API main page design?
Each endpoint would have its own page that can be accessed from the sidebar or clicking any of the description headings.

image

image

The actual docs of the endpoint is just a placeholder for now that I copied from the old page.

**Comment by geoffrey-wu** [comment]
Looks perfect to me!

File: server/schemas.js
Comment (g3ner1c) [inline server/schemas.js:1]: So these schemas I wrote in accordance with the latest json schema specs, but [mongo appears to use an older incompatable spec.](https://www.mongodb.com/docs/manual/reference/operator/query/jsonSchema/#syntax)

It shouldn't be too much trouble converting to the older spec for mongo, but I'm not very familar with mongo at all so I don't really know how implementing schemas there would work.

**Comment by g3ner1c** [comment]
quick question: is it stylized "QBReader", "QB Reader", "qbreader", or whatever else lol

**Comment by geoffrey-wu** [comment]
> quick question: is it stylized "QBReader", "QB Reader", "qbreader", or whatever else lol

I would go for QB Reader, but in honesty I'm not sure if anyone knows.

**Comment by geoffrey-wu** [comment]
Any idea on when this will be completed? It looks very exciting :eyes:

**Comment by g3ner1c** [comment]
got caught up in school will probably be done later this week!

**Comment by g3ner1c** [comment]
I've gone back and checked out the up to date `scss` files and integrated any additions to the docs in `main`. It should be ok to override conflicts with changes in this pr.

**Review by g3ner1c — COMMENTED** [review]
I didn't include URL examples for requests, because I feel like that sort of thing can be easily done with tools like [HTTPie](https://httpie.io/), especially their [web app](https://httpie.io/app). I'll definitely consider adding things like cookbooks and quickstarts if the API gets large enough though.

**Comment by g3ner1c** [comment]
# Summary

## Changes
- Revamped API documentation with seperate pages for each endpoint, schema specs, and detailed typing information
- Sidebar to traverse docs ([I made it responsively disappear on small enough windows but I haven't tested it on mobile](https://github.com/qbreader/website/pull/150/files#diff-b4fc7472a1af3b5aaa37faa1dbc8226b8e70d9a9a73b39b600a7a06c4b5c4d1bR76))
- `/api-info` is now `/api-docs`, requests will be redirected to not break hyperlinks
- Added JSON schemas in `server/schemas.js`, they kinda just vibin there rn not really sure how they'll be implemented

## Notes and ideas for the future
- Some of the endpoints return full objects while others only return arrays. I think it might be worth converting one that don't return objects (like `api/check-answer`) into returning objects for consistency and easier use for developers.
- Many questions in the database are missing certian fields in the schema like `formatted_answer`, could definitely be a problem for people expecting all fields to be complete.
- Quickstarts and cookbooks
- probably should change `api/random-question` to `GET` soon
- accordion design???

**Pull request merged**